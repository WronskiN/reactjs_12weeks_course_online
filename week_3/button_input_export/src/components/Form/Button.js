import React from "react";

function Button(props) {
  const { label, bgColor, color } = props;

  const styles = {
    backgroundColor: `${bgColor}`,
    color: `${color}`,
    fontSize: 16,
    border: "none",
    borderRadius: 5,
    padding: 10,
    letterSpacing: 1,
  };

  return <button style={styles}>{label}</button>;
}

Button.propsDefault = {
  label: "click me",
  bgColor: "darkBlue",
  color: "white",
};

export default Button;
