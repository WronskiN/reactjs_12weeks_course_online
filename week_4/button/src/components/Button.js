import React, { useRef, useEffect } from 'react';

export default function Button() {
  const buttonChange = useRef();

  useEffect(() => {
    setTimeout(() => {
      buttonChange.current.style.color = '#fff';
      buttonChange.current.style.backgroundColor = 'red';
    }, 500);
  });
  return <button ref={buttonChange}>Primary</button>;
}
