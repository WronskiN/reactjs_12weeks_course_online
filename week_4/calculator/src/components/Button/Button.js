import React from 'react';
import './style.css';

function Button({ type, text, click }) {
  return (
    <button type={type} onClick={click} className="btn">
      {text}
    </button>
  );
}

export default Button;
