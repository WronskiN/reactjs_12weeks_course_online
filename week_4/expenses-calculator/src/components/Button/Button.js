import React from 'react';

function Button({ children, className }) {
  return <button className={`buttonDefault ${className}`}>{children}</button>;
}

export default Button;
