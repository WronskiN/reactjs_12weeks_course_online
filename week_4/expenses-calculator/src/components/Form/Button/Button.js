import React from 'react';

function Button({ value }) {
  return <button className="button">{value}</button>;
}

export default Button;
