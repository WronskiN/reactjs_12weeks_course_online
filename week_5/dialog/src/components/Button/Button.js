import React from 'react';
import './style.css';

function Button() {
  return <button>Open Dialog</button>;
}

export default Button;
