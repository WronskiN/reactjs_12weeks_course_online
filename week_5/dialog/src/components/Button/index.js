export { default } from './Button';
export { default as ButtonWrapper } from './ButtonWrapper';
