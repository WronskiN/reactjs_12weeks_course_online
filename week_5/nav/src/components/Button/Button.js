import React from 'react';

function Button(props) {
  const { content, btnClass, handleClick } = props;
  return (
    <button className={btnClass} onClick={handleClick}>
      {content}
    </button>
  );
}

export default Button;
