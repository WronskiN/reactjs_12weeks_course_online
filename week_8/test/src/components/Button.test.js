import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Button from './Button';

it('renders ok', () => {
  const tree = renderer.create(<Button label='Click me!' />).toJSON();
  expect(tree).toMatchSnapshot();
});

// describe('testing with jest', () => {
//   it('should isOk to be true', () => {
//     const isOk = true;
//     expect(isOk).toBe(true);
//   });
//   it('should array length to be 2', () => {
//     const names = ['Norbert', 'Justyna'];
//     expect(names.length).toBe(2);
//   });
// });

describe('Button component', () => {
  it('should renders OK 0', () => {
    const wrapper = shallow(<Button />);
    const button = wrapper.find('button');
    expect(button.text()).toBe('OK 0');
  });
  it('should renders Click me 0', () => {
    const wrapper = shallow(<Button label='Click me' />);
    const button = wrapper.find('button');
    expect(button.text()).toBe('Click me 0');
  });
  it('should have counter set to 0', () => {
    const wrapper = shallow(<Button />);
    const counter = wrapper.find('button span');
    expect(counter.text()).toBe('0');
  });
  it('should increment counter', () => {
    const wrapper = shallow(<Button />);
    const button = wrapper.find('button');
    let counter = wrapper.find('button span');
    expect(counter.text()).toBe('0');
    button.simulate('click');
    counter = wrapper.find('button span');
    expect(counter.text()).toBe('1');
  });
});
