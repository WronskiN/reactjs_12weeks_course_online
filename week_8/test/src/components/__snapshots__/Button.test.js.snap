// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ok 1`] = `
<button
  onClick={[Function]}
>
  Click me!
   
  <span>
    0
  </span>
</button>
`;
